lib/wadler/print.rb

```diff
@@ -39,43 +39,41 @@ def output
     # @return [Nil]
     def group(indent = 0, open_obj = '', close_obj = '',
               break_type = Oppen::Token::BreakType::CONSISTENT, &)
+      tokens <<
+        case break_type
+        in Oppen::Token::BreakType::CONSISTENT
+          Oppen.begin_consistent(offset: indent)
+        in Oppen::Token::BreakType::INCONSISTENT
+          Oppen.begin_inconsistent(offset: indent)
+        end
+
       if !open_obj.empty?
         self.break
         text(open_obj)
       end
 
       nest(indent, break_type, &)
 
-      if !close_obj.empty? # rubocop:disable Style/GuardClause
+      if !close_obj.empty?
         self.break
         text(close_obj)
       end
+
+      tokens << Oppen.end
     end
 
     # @param indent [Integer] nest indentation
     # @param break_type [Oppen::Token::BreakType] nest breaking type
     #
-    # @yield the block of text that requires nesting
-    #
     # @return [Nil]
-    def nest(indent = 2, break_type = Oppen::Token::BreakType::CONSISTENT, &block)
-      previous_indent = current_indent
-      @current_indent = indent
-
-      tokens <<
-        case break_type
-        in Oppen::Token::BreakType::CONSISTENT
-          Oppen.begin_consistent(offset: indent)
-        in Oppen::Token::BreakType::INCONSISTENT
-          Oppen.begin_inconsistent(offset: indent)
-        end
+    def nest(indent, break_type = Oppen::Token::BreakType::CONSISTENT, &block)
+      @current_indent += indent
 
       raise LocalJumpError if !block_given?
 
       block.call
 
-      tokens << Oppen.end
-      @current_indent = previous_indent
+      @current_indent -= indent
     end
```

Comment:
That might be dangerous:

On one hand, we're advancing `@current_indent`, and then we might raise an exception.
When that happens, we don't decrease it back again, and we're in a bad state.

Use [`ensure`](https://ruby-doc.org/docs/ruby-doc-bundle/UsersGuide/rg/ensure.html)